daemon/stop.go

```diff
@@ -78,12 +78,16 @@ func (daemon *Daemon) containerStop(container *containerpkg.Container, seconds i
 		logrus.Infof("Container %v failed to exit within %d seconds of signal %d - using the force", container.ID, seconds, stopSignal)
 		// 3. If it doesn't, then send SIGKILL
 		if err := daemon.Kill(container); err != nil {
-			// Wait without a timeout, ignore result.
-			<-container.Wait(context.Background(), containerpkg.WaitConditionNotRunning)
```

Comment:
I wonder if this wait is even needed. What's the reasoning behind waiting indefinitely if `Kill` returns an error? 
It seems this code assumes that there will be an `containerpkg.WaitConditionNotRunning` even if `Kill` fails, but that doesn't seem to be true in all cases (if at all), which is why this function waits for ever, as described in https://github.com/moby/moby/issues/41579

I think we should just return the error here
Thoughts?

Reply:
I think it's worth waiting.

Also, FWIW, I think the potential for deadlock here is small.
There were some bugs related to error handling from runc as well as event processing from containerd that would trigger a problem here, but those are cleaned up.

That said, totally fine to not wait for ever.
The assumption to wait forever has likely been passed down from old versions where docker directly managed the container process, where as now that has containerd in the middle.

Reply:
I agree that the potential for deadlock is small, but if docker kill is failing for some reason then this does currently leak goroutines, assuming whoever is using the client library times out at some point and retries (as is the case for the ecs agent)

Reply:
I'm also not sure about the usefulness of waiting, @cpuguy83 would you be opposed to at least reducing the timeout to 2 seconds like the code does in a few other places?